tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,289 @@
+	}
+	return awsProvider
+}
 
-	err = cloud.CreateWindowsVM()
+// TestCreateWindowsInstance creates a Windows instance on AWS and assert no error from instance creation.
+// If it fails, abandon the rest of the test and clean up.
+func TestCreateWindowsInstance(t *testing.T) {
+	err := awsProvider.CreateWindowsVM()
 	assert.NoError(t, err, "error creating Windows instance")
 	if err != nil {
 		// TODO: Remove the lingering resources as part of test setup.
-		t.Fatalf("Error while creating a VM delete the virtual machine")
-	}
-	// Type Assert to AWS so that we can test other functionality
-	aws, ok := cloud.(*awscp.AwsProvider)
-	if !ok {
-		t.Fatal("Error asserting cloudprovider to aws")
+		assert.FailNow(t, "Error while creating a VM delete the virtual machine")
 	}
+}
+
+// TestInstallerJsonFileRead reads instance and security group IDs from windows-node-installer.json file and
+// updates the createdInstanceID and CreatedSgID global variables.
+// If it fails, abandon the rest of the test and clean up.
+func TestInstallerJsonFileRead(t *testing.T) {
 	// Check instance and security group information in windows-node-installer.json.
 	info, err := resource.ReadInstallerInfo(dir + "/" + "windows-node-installer.json")
 	assert.NoError(t, err, "error reading windows-node-installer.json file")
 	assert.NotEmpty(t, info.SecurityGroupIDs, "security group is not created")
 	assert.NotEmpty(t, info.InstanceIDs, "instance is not created")
-	instance, err := aws.GetInstance(info.InstanceIDs[0])
-	if err != nil {
-		t.Fatalf("error getting ec2 instance object from instance ID %v", info.InstanceIDs[0])
+
+	if len(info.InstanceIDs) < 1 || info.InstanceIDs[0] == "" || len(info.SecurityGroupIDs) < 1 || info.
+		SecurityGroupIDs[0] == "" {
+		// TODO: Remove the lingering resources as part of test setup.
+		assert.FailNow(t, "error reading from windows-node-installer.json file")
 	}
-	infraID, err := aws.GetInfraID()
+
+	// Update instance and sg IDs.
+	createdInstanceID = info.InstanceIDs[0]
+	createdSgID = info.SecurityGroupIDs[0]
+}
+
+// TestInstanceRetrievalAfterCreation updates the createdInstance global object for subsequent tests.
+// If it fails, abandon the rest of the test and clean up.
+func TestInstanceRetrievalAfterCreation(t *testing.T) {
+	instance, err := awsProvider.GetInstance(createdInstanceID)
 	if err != nil {
-		t.Fatalf("error while getting infrastructure ID for the given OpenShift cluster")
+		// TODO: Remove the lingering resources as part of test setup.
+		assert.FailNow(t, fmt.Sprintf("error getting ec2 instance object, %v", err))
 	}
-	if !checkIfInfraTagExists(instance, infraID) {
-		t.Fatalf("Infrastructure tag not available for %v", info.InstanceIDs[0])
+	assert.NoError(t, err, "error getting ec2 instance object")
+	createdInstance = instance
+}
+
+// TestInstanceIsRunning asserts if an instance is in the running state.
+// Instance input needs to be updated before use.
+func TestInstanceIsRunning(t *testing.T) {
```

Comment:
These one line asserts need to be part of your main test. You are actually not running any tests here.